app/featurelayerpair.cpp

```diff
@@ -60,10 +60,12 @@ bool FeatureLayerPair::hasValidGeometry() const
 {
   Q_ASSERT( mLayer );
 
-  if ( !mFeature.hasGeometry() )
-    return false;
+  bool isNoGeometryLayer = mLayer->geometryType() == QgsWkbTypes::NullGeometry;
```

Comment:
I would suggest to use `mLayer->isSpatial()` - its body allows both NullGeometry and UnknownGeometry.